saw-core-coq/src/Verifier/SAW/Translation/Coq/SpecialTreatment.hs

```diff
@@ -605,3 +610,27 @@ Fixpoint zip (a b : sort 0) (m n : nat) (xs : Vec m a) (ys : Vec n b)
   end
 .
 |]
+
```

Comment:
Why are these included as strings in `SpecialTreatment.hs`, instead of as proofs in `SAWCoreScaffolding.v` or some other Coq support library file? That would make these definitions less brittle and easier to modify in the future.

Reply:
`SAWCoreScaffolding` would seem like a natural home for these, but there is one complication: the types of `{at,gen}WithProof` depend on `IsLtNat`, which is an auto-generated definition in `SAWCorePrelude`. Since `SAWCorePrelude` imports `SAWCoreScaffolding`, trying to put `{at,gen}WithProof` in `SAWCoreScaffolding` would induce an import cycle.

Another alternative that I considered is putting these functions in `SAWCorePreludeExtra`. That doesn't work either, unfortunately, since there are other definitions in `SAWCorePrelude` that are defined in terms of `{at,gen}WithProof`, and we can't import `SAWCorePreludeExtra` in `SAWCorePrelude` (again, import cycles).

The treatment here is the only way that I could make things work given those constraints. If you find this bothersome, we could also consider giving a special translation from `IsLtNat` to a definition in `SAWCoreScaffolding` (say, `Definition IsLtNat := @lt`). Let me know what you would prefer.

Reply:
Yeah, I figured it was some sort of complicated dependencies like that. I think your idea about defining `IsLeNat := @le` and `IsLtNat := @lt` makes a lot of sense.

Reply:
`SAWCoreScaffolding.v` now contains `IsLtNat`, and `SAWCoreVectorsAsBitvectors.v` now contains `{at,gen}WithProof`.

Reply:
Weird that `IsLeNat` is already wired in but `IsLtNat` isn't. Hmm...